### ISSUE TITLE
Only works with v1 (jar) signing

### ISSUE THREAD

**Issue by swilson** [DimaKoz/stunning-signature#11]
As it relies on the certificate being present in the META-INF directory. Also note that Android Studio will no longer insert the V1 signature when launching debug builds, so the check needs to be disabled in this situation.

https://stackoverflow.com/questions/57584467/cert-rsa-is-missing-in-debug-build-apk-file-after-upgrade-androidstuio-and-gradl

**Comment by DimaDDM** [comment]
Hello. Did you find a solution for version 2.0? Looks like **v1SigningEnabled true** not working anymore for minSDK 26+

**Comment by dquang0706** [comment]
Did anyone find a solution for version 2.0, 3.0, ...?

Many thanks,

**Comment by Hyupai** [comment]
Thats true. Its support only v1 sign... i dont find a solution for v2 and v3, but we can make  a hash of all files inside apk, perhaps its can be a good idea... get the Fields and Methods of a class can be a useful way too

**Comment by DimaDDM** [comment]
If anyone has a solution, an example, or an idea to protect the application for new versions, please share. It seems to me that hashing all files is not the most convenient option, although I also thought about it.

**Comment by Hyupai** [comment]
@DimaDDM In fact, hashing all the files can be expensive depending on the size of the apk, but it may be necessary.

I was successful in putting v1 downgrade to min sdk 23 (Android 6.0), in my case this did not interfere with my functions.

After the downgrade I inserted:

 signingConfigs {
        mySigningConfig {
            keyAlias ​​'theAlias'
            keyPassword 'thePassword'
            storeFile file('theKeystore.jks')
            storePassword 'thePassword'
            v1SigningEnabled true
            v2SigningEnabled true
        }
    }

With this my app was compiled with the V1 and V2 signature schemes. I tested with one of the signature killers (NP Manager) and the native method actually returns true. I hope this helps you. I also checked the methods and fields of my classes in the APK, since it is very small.
